### ISSUE TITLE
Review the gRPC-Web protocol proposal 

### ISSUE THREAD

**Issue by wenbozhu** [grpc/grpc#8682]
[Link to the protocol spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md)

Please post your review comments here.

**Comment by wenbozhu** [comment]
@yugui

**Comment by zinuga** [comment]
@mattklein123 have a look

**Comment by louiscryan** [comment]
@wenbozhu 
@ejona86 
@ctiller 

For 
"Response status encoded as part of the response body
Key-value pairs formatted as HTTP/1.1 headers block (without the empty newline \r\n to terminate the block)"

Why the HTTP format rather than HTTP2 literal format which would be faster to parse. We can also allow binary headers here without specific encoding.

**Comment by lhecker** [comment]
I hope I'm not intruding with my following naive question, since it appears that this issue is only intended for a purely technical discussion:

Will _full_ (Bidi-)streaming be supported? Because it only says this:

> 1. support any HTTP/*

and

> Pending on whatwg fetch/streams to be finalized and implemented in modern browsers

which sounds like a "no" and that would be kinda disheartening. ☹
If that's correct I'd be very interested in turning this into a "yes", especially since I expect Websockets (which offer that) to be more widely supported right now than fetch/streams might be in 1-2 years.

**Comment by mattklein123** [comment]
This looks good to me. This would allow us to get rid of the Envoy bridge filter hack if the protocol was widely supported.

**Comment by wenbozhu** [comment]
@lhecker There is no plan to support websockets. The biggest issue with WS is that the client library still needs to fall back to HTTP. 

Server-streaming is sufficient for most use cases, e.g. when (simpler) unary calls are prohibitively expensive or slow.

**Comment by wenbozhu** [comment]
@louiscryan Agreed that the http/1.1 encoding of trailers is insufficient or desired. Will fix the spec..

**Comment by lhecker** [comment]
@wenbozhu I'm not sure if I understand this:

> The biggest issue with WS is that the client library still needs to fall back to HTTP.

Did you mean that an unary RPC call should directly map to a single HTTP request or are you referring to the fact that Websockets aren't widely supported?

> Server-streaming is sufficient for most use cases

So no Bidi-Streaming then? (Sorry for the repeated question.)

I hope I misunderstood but if not I'm sure that many others besides me would find it a bit frustrating if streaming doesn't work in its entirety.

**Comment by wenbozhu** [comment]
@lhecker 

> Did you mean that an unary RPC call should directly map to a single HTTP request or are you referring to the fact that Websockets aren't widely supported?

The latter, due to proxies/firewalls etc.

> So no Bidi-Streaming then?

Correct.

**Comment by lhecker** [comment]
@wenbozhu

> > So no Bidi-Streaming then?
> Correct.

Thanks! That's just what I wanted to know.

So what's your stance on letting the open source community _decide_ what they want? Up till now I really get the stereotyped feeling of "Google wants this and that's how it's going to be". I hope I'm wrong again. 😞

**Comment by lhecker** [comment]
In fact I wonder why we can't have both by transparently using regular HTTP requests (like in this proposal) wherever we can and Websockets where streaming is required.

I don't believe anyone would be surprised by gRPC requiring Websockets and similar proxy-incompatible technologies for Bidi-streaming where proxy-incompatibility is implied anyways already. But an immature solution for simplicities and compatibilities sake? That just screams that people will build their own solution sooner or later and I don't see why that should be any good or even needed. 😕

**Comment by wenbozhu** [comment]
@lhecker  Thanks for the feedback!

The grpc-web spec introduces minimum changes to the existing grpc over HTTP/2 protocol. The new spec itself is bidi ready, while the actual bidi support is subject to environment/platform/network constraints (same for the grpc or WebSocket, as you noted).

How about we open a separate issue to discuss how to support bidi streams with grpc-web clients, after this spec has been reviewed and implemented? Your contribution in this work or any future work to support bidi streams is more than welcomed.  @zinuga

**Comment by lhecker** [comment]
> How about we open a separate issue to discuss how to support bidi streams with grpc-web clients, after this spec has been reviewed and implemented?

Although I believe not having such a support from the get-go would be a real shame I can kinda understand that. Having a separate issue then would be more than welcomed. Do you mind opening one? 🙂 Again I'm sorry for stirring up this issue so much. I'm definitely looking forward on helping out designing and implementing such an extension. 🙂

**Comment by wenbozhu** [comment]
Thanks for all the feedback. 

Plan to close the issue in a week or so. @yugui (to have a look at this spec soon ;)

**Comment by yugui** [comment]
LGTM

**Comment by wenbozhu** [comment]
Thanks for all the feedback. 

Implementation of the new spec will happen in [grpc/grpc-web](https://github.com/grpc/grpc-web)

Ping @zinuga if you need early access to grpc-web.

**Comment by wenbozhu** [comment]
@lhecker 

https://github.com/grpc/grpc-web/issues/24  (bidi support discussion)

**Comment by rutsky** [comment]
@wenbozhu where did https://github.com/grpc/grpc-web/ gone?

**Comment by wenbozhu** [comment]
@rutsky It's an early access project. Just white-listed your user id.

**Comment by triptec** [comment]
@zinuga could I get read permission to grpc-web?

**Comment by atishpatel** [comment]
@zinuga Could I please get early access to grpc-web?

**Comment by wenbozhu** [comment]
Done. Please check.

**Comment by wenbozhu** [comment]
@triptec @atishpatel

**Comment by arbesfeld** [comment]
@wenbozhu I would also like early access please :)

**Comment by Nishant-Pathak** [comment]
Please provide to me too. Github username - Nishant-Pathak

**Comment by fire** [comment]
Maybe some sort of request form would be useful. I would also like to review it.

**Comment by paramr** [comment]
Would love to try out grpc-web too.  Really looking forward for bidi streaming.  Real time updates are kind of expected for interactive apps now a days.

**Comment by ericbets** [comment]
@wenbozhu I'll jump on this bandwagon. I hereby request white listing.

**Comment by xealot** [comment]
@wenbozhu I love a good bandwagon

**Comment by mxab** [comment]
@wenbozhu since it seems to be super secret may I also have a look?

**Comment by namtzigla** [comment]
@wenbozhu may I have access to grace-web, please !

**Comment by wenbozhu** [comment]
Done. Feedback is welcome ..

**Comment by paramr** [comment]
@wenbozhu Thanks for responding but I dont see grpc-web yet.

**Comment by fire** [comment]
@wenbozhu I don't see it either.

**Comment by christianpv** [comment]
@wenbozhu I would like to try it.

**Comment by meastp** [comment]
@wenbozhu I would love to have a look as well, please

**Comment by miofthena** [comment]
@wenbozhu  Add me too, please

**Comment by tukafab** [comment]
@wenbozhu can you add me to pls:)

**Comment by prithivraj** [comment]
Could you give me access to the beta too? Thanks. @prithivraj

**Comment by wenbozhu** [comment]
DONE. We plan to publish the grpc-web rep in Q1.

Thanks for your interests.

**Comment by ramasubiah** [comment]
Could you please give me beta access ?

**Comment by andreabreu-me** [comment]
@wenbozhu Add me too please

**Comment by CooCooCaCha** [comment]
Me too please!

**Comment by sampleapp** [comment]
Why not me! pls pls.

**Comment by RuairiSpain** [comment]
Can I get access to the grpc-web spec. Thank you.

**Comment by wenbozhu** [comment]
DONE

**Comment by pcj** [comment]
@wenbozhu Please add me as well to whitelist.  Thanks.

**Comment by rms1000watt** [comment]
@wenbozhu Count me in! Thanks a ton!

**Comment by quiasmo** [comment]
Would like access as well. Did fill out some form somewhere for access but haven't heard anything.

**Comment by wenbozhu** [comment]
DONE

**Comment by rolandhawk** [comment]
@wenbozhu please add me to whitelist. Thank you very much.

**Comment by arthur-tacca** [comment]
@wenbozhu Could I please be added too? I would be happy to test it. Many thanks.

**Comment by ZachBray** [comment]
@wenbozhu please can you add me too? Thank you.

**Comment by PotterDai** [comment]
@wenbozhu  Add me pls. Thank you!

**Comment by imysak** [comment]
@wenbozhu can you add me too? Thank you.

**Comment by peterrosell** [comment]
I'm also interested in this. Can you add me? Thank you.

**Comment by joa** [comment]
@wenbozhu Kindly add me as well please. Thank you.

**Comment by maehld** [comment]
@wenbozhu I'm currently evaluating gRPC and would like to take a look at the grpc-web repository. Thanks.

**Comment by jeffomatic** [comment]
@wenbozhu Could I please be added to the whitelist? Thanks.

**Comment by marsbroshok** [comment]
@wenbozhu Could you kindly give me access to grpc-web please? Thanks!

**Comment by wenbozhu** [comment]
DONE

**Comment by rmhrisk** [comment]
@wenbozhu could you please give me access as well.

**Comment by danbenedicto** [comment]
@wenbozhu Could you please add me too? Thank you!

**Comment by pilt** [comment]
@wenbozhu Dito. ;)

**Comment by traceon** [comment]
@wenbozhu add me too, please. Thank you!

**Comment by widmogrod** [comment]
@wenbozhu could you please give me access as well

**Comment by c4milo** [comment]
@zinuga, @wenbozhu I'm also interested in looking into grpc-web! Thanks!

**Comment by xiang90** [comment]
I am interested in grpc-web.

**Comment by arbesfeld** [comment]
Me too, please

**Comment by simonhorlick** [comment]
I'm also interested in grpc-web. Could you please add me?

**Comment by mjkim** [comment]
Could you please give me access grpc-web?

**Comment by caseylucas** [comment]
can you add me please?

**Comment by luckystar513** [comment]
@zinuga could I get read permission to grpc-web?

**Comment by wenbozhu** [comment]
DONE

**Comment by public** [comment]
@zinuga @wenbozhu Can I have access to grpc-web please?

**Comment by codesuki** [comment]
If possible I would like to have access, too. To test it.

**Comment by wenbozhu** [comment]
DONE

**Comment by nevi-me** [comment]
@wenbozhu May I please also have read-only access to grpc-web?

**Comment by iheanyi** [comment]
@wenbozhu Could I have access to `grpc-web` to test as well? I just got `grpc` working with [Electron](https://electron.atom.io), would love to see what's going on with `grpc-web`.

**Comment by samiur** [comment]
@wenbozhu would love access to grpc-web also!

**Comment by juanpmarin-infocorp** [comment]
@wenbozhu Could I have access to grpc-web ?
Thanks !

**Comment by wenbozhu** [comment]
DONE

@iheanyi  I am new to Eletron .. does the client run as a chrome app?

**Comment by holgerkoser** [comment]
@wenbozhu Is it possible to get read access to grpc-web?
Thanks!

**Comment by orjanbruland** [comment]
@wenbozhu would it be possible to get read access to grpc-web? Thanks!

**Comment by iheanyi** [comment]
@wenbozhu Yes it does, at least it runs using V8 to my knowledge. It's pretty dope. But it uses Node.js under the hood when it comes to the gRPC client, which I find to be promising. I am not sure how it'd work with things being redistributed as an application.

**Comment by nevi-me** [comment]
@iheanyi re. redistribution; it should work properly as it does on a server. The main thing to consider is disconnect/reconnects as a server will typically be expected to be online all the time, while a desktop app might go offline for a while. This mainly affects streaming endpoints which when `onError` is called.

**Comment by iheanyi** [comment]
@nevi-me Solid point to consider. I'm working on a proof of concept application [here ](https://github.com/iheanyi/go-electron-grpc) of this entire integration (Go +gRPC + Electron). Your mentioning of streaming makes me want to see if I can tweak this example to be a chatroom example because I think that it would be a good proof of concept for demonstrating "streaming" of between the client and server. But hey, I'm new to this, so I may be mistaken, haha.

**Comment by nevi-me** [comment]
@iheanyi just looked at your project, the stuff to really test is streaming, especially bi-directional. Unary calls are sort of a "given" at this point as they're the easier of the 2.

I'm using gRPC on connecting microservices together at work, Mostly between Node.js and Apache Spark (I find PySpark better that ScalaPB). On personal projects I've got client/server instances in Java, Node.js and Python. The Node.js client connects to a constant bi-directional streaming endpoint on the Java side. So I have lifecycle calls on both ends to reconnect when necessary.

On the desktop you won't have the luxury of constantly trying to reconnect as that'll impact CPU consumptions and all the perils that follow. **If you wanna continue talking, we can move elsewhere so we don't keep spamming the project.**

**Comment by ankoh** [comment]
@wenbozhu I would also love to take a look at how you are going to implement grpc-web.

**Comment by jimbojetlag** [comment]
@wenbozhu would you please add me to grpc-web too.

**Comment by tejasmanohar** [comment]
@wenbozhu Could I have access to grpc-web? We've been using GRPC in production for almost a year across Go, Python, and Node, and I'd love to take a look, provide feedback, etc.

**Comment by wenbozhu** [comment]
DONE

**Comment by IvanSafonov** [comment]
@wenbozhu could you please add me too? Thanks!

**Comment by mwitkow** [comment]
@wenbozhu We did something similar to what you're proposing in, and the proof of concept is in:
https://github.com/mwitkow/grpc-browser-compat

Basically it's a Go proxy + browser polyfill for the NodeJS gRPC transport. 

Can you please add me to the grpc-web repo? :) Thanks!

**Comment by duhruh** [comment]
@wenbozhu can i please get added to grpc-web 😄

**Comment by gabriel** [comment]
@wenbozhu can I also get added to grpc-web?

**Comment by colonelmo** [comment]
Can I also get early access to the project?

**Comment by meekapps** [comment]
Considering a move to gRPC in production (from a custom serialization scheme/RPC) but not having a web client is a limiting factor. How far off is a grpc-web general release? Is it a web sockets adapter or something else? Is it possible to get read access to grpc-web project?

**Comment by wenbozhu** [comment]
DONE

@meekapps  It's production ready but we may change the protocol/apis etc.

Read the doc/demo and let us know your feedback.

===

@mwitkow 

> browser polyfill for the NodeJS gRPC transport

what's the actual transport the polyfill speaks.. . it can't be the actual grpc transport, can it?

**Comment by Chumper** [comment]
I will patentially wait for you to make the repo public. Would love to try it before I write a separate service just for the conversion from grpc to websocket.

**Comment by chrisrcoles** [comment]
@wenbozhu can I also get added to grpc-web please?

**Comment by mackross** [comment]
Me too please

**Comment by dylanbutler** [comment]
@wenbozhu me too please? :)

**Comment by mwitkow** [comment]
@wenbozhu It is*almost* the gRPC transport.
It uses Fetch API (with fallback using [chunked-request](https://github.com/jonnyreeves/chunked-request)), the same binary framing of the gRPC transport. The only difference is that trailers (grpc status and desc) are not sent as a HEADER frame but as a DATA frame encoded in a [terminator.proto](https://github.com/mwitkow/grpc-browser-compat/blob/master/terminator.proto). It's a proof of concept, but seems to work and we'll try it out for some of our services.

**Comment by redaready** [comment]
@wenbozhu Can I also get early access to the project? Thanks

**Comment by capone212** [comment]
@wenbozhu early access, please! thanks!

**Comment by yongtang** [comment]
@wenbozhu Can I also have the early access to the project if possible? Thanks

**Comment by wenbozhu** [comment]
DONE (accesses granted)

**Comment by zhoshen** [comment]
@wenbozhu  Can I also get early access to the project? Thanks

**Comment by chena** [comment]
@wenbozhu Can I also get early access to `grpc-web`? Thank you!

**Comment by x0d3r** [comment]
@wenbozhu early access, please !

**Comment by sulliwane** [comment]
@wenbozhu Can I also get early access to grpc-web? Thank you!

**Comment by lchenn** [comment]
Can I get early access to grpc-web please?

**Comment by dobegor** [comment]
@wenbozhu I know it may be very annoying... could I please have a read access to grpc-web, too?
Thanks.

**Comment by kangjoni76** [comment]
@wenbozhu , Can I get early access please.. thanks

**Comment by LeeWong** [comment]
@wenbozhu , Can I get an early access as well pls? It is very exciting news.

**Comment by nwong4932** [comment]
@wenbozhu, Can I get early access please.. thanks

**Comment by nicksrandall** [comment]
@wenbozhu I would like to test and give feedback on this. Could I please get early access? Thanks!

**Comment by lorenz** [comment]
@wenbozhu I would like to try grpc-web too, can I get early access?

**Comment by citizenrich** [comment]
@wenbozhu jumping on this train too. would love early access.

**Comment by thearchitect** [comment]
@wenbozhu Can I get early access too, plz?

**Comment by IceMan81** [comment]
@wenbozhu Would like to get early access to grpc-web and gice feedback. Thx

**Comment by tobiaskohlbau** [comment]
@wenbozhu would like to jump on the train, too.
Possible to get some early accesss goodness?

**Comment by jcarrig** [comment]
@wenbozhu, my team is evaluating using grpc in prod, but we also need a web component.  If possible, I would like access to grpc-web.  Thanks!

**Comment by lr4** [comment]
@wenbozhu Could I get early access to the grpc-web as well please? Thanks.

**Comment by wenbozhu** [comment]
DONE

**Comment by jls5177** [comment]
@wenbozhu Could I also be added to the early access list? Thanks in advance.

**Comment by ryanstinson** [comment]
@wenbozhu Early access please? Thanks

**Comment by kkimdev** [comment]
@wenbozhu I would like to request early access. Thank you.

**Comment by zxr90** [comment]
@wenbozhu May I have read-only access to grpc-web please? Evaluating this for my company. Thank you!

**Comment by UnnoTed** [comment]
@wenbozhu Hey, would you please add me to the list? thanks.

**Comment by mustafaakin** [comment]
@wenbozhu Can you add me to list? Thanks

**Comment by cube2222** [comment]
@wenbozhu Please add me to the early-access list too.

**Comment by PranavSathy** [comment]
Can I please be added as well :)

**Comment by mably** [comment]
@wenbozhu I am working on an Electron gRPC app, could I get early access to grpc-web please ?

**Comment by juscamarena** [comment]
Second that ^

**Comment by theill** [comment]
@wenbozhu please whitelist me too in your next batch. thanks!

**Comment by sayrer** [comment]
Hi, I'd like access to the grpc-web repo as well. Thank you!

**Comment by brianstarke** [comment]
This sounds awesome, I'd love early access as well.  Thank you!

**Comment by jmzwcn** [comment]
@wenbozhu ,
Hi, I am investigating how to handle GRPC-web related issue. Could you please add me to access list?

**Comment by wenbozhu** [comment]
DONE.

**Comment by wuman** [comment]
Hi @wenbozhu we are using grpc extensively in our backend and would like to bring it to the browser.  Could you please add me to the early access whitelist?

**Comment by PeerXu** [comment]
Hi @wenbozhu , Could you add me to the early access list? Thanks.

**Comment by melindalu** [comment]
@wenbozhu I'd love to get access as well — thanks!

**Comment by masonk** [comment]
@wenbozhu, access please? (Working a Postgres schema -> gRPC generator, want to test it for gRPC-Web as well).

**Comment by cdosoftei** [comment]
@wenbozhu I would like to request grpc-web early access. Thanks in advance!

**Comment by MidnightWonderer** [comment]
What happened to this form
https://docs.google.com/forms/d/15iRDHoP-VBenc4hWgKn7bk7IirJLgs0uh88nw1vi_Hc/viewform?edit_requested=true
![Screenshot](http://i.imgur.com/JotCmVj.png)
Does it get hacked?
Is it still usable?

**Comment by crotger** [comment]
I'm a teaching assistant for a distributed systems class that's using GRPC and I'd love to check out the grpc-web repo!

**Comment by donatasm** [comment]
@wenbozhu would like to access to grpc-web too. Thanks

**Comment by ottob** [comment]
@wenbozhu I'm curious about grpc-web too. Is it possible to get access?

**Comment by RealDude** [comment]
@wenbozhu Please give me access.

**Comment by virusdave** [comment]
@wenbozhu Hi!  Evaluating gRPC, with browser support being a huge factor.  Would appreciate early read access to grpc-web as well.  Thanks in advance!

**Comment by wenbozhu** [comment]
DONE

@MidnightWonderer  The sign-up form is still alive, to collect general interests in grpc and grpc-web. The fact that this issue thread becomes the defacto sign-up sheet wasn't initially expected :)

Cheers.

**Comment by rslonik** [comment]
@wenbozhu I would like to test grpc-web if possible, thanks in advance

**Comment by nukr** [comment]
May I have grpc-web early access please ?

**Comment by ddevienne** [comment]
@wenbozhu What's the point of not having the repo public and that sign-up form, really? Interest in a gitHub project is easily gaged via stars and forks, no need for forms. What are you afraid of, not doing this work in public? Such pseudo secrecy is surprising / unexpected / worrying...

**Comment by ianha** [comment]
@wenbozhu Please add me for early access, thanks.

**Comment by lukeblechl** [comment]
@wenbozhu, whitelist for me as well. I'm with @ddevienne, I say just make it public, even if it's really rough around the edges. I've got your back if someone starts complaining. :)

**Comment by yves-pkstd** [comment]
I am interested in early access as well please.  Thanks.

**Comment by deejross** [comment]
@wenbozhu I am also interested in early access. Thank you in advance!

**Comment by jezell** [comment]
@wenbozhu I'm also interested in early access!

**Comment by kenkeiter** [comment]
@wenbozhu Would love access as well, if possible.

I don't think the lack of openness here is surprising, unexpected, or worrying -- if you're inventing something like a new protocol, the last thing you want is for people to begin adopting it (and relying on its conventions, idiosyncrasies, etc.) before they've been finalized. This is exactly how you end up in the poor situation of having to compromise your product 

You can slap hundreds of warnings on something like this saying "EXPERIMENTAL – USE AT YOUR OWN RISK" and people will still find a way to blame you when they have to upgrade and overhaul large parts of their system to make room for an actual _released_ version of the tool.

There's a difference between openness and transparency. How the maintainers choose to conduct themselves in this context is their decision alone. They're being generous enough to do this work -- when they release this work is up to them.

**Comment by pkieltyka** [comment]
yea, Im wondering too what is the status of grpc-web? I believe it was originally set for a public release as beta for Q1 of 2017, hopefully thats still on-target. At least to reveal the repo for public consumption and feedback to get things moving faster

**Comment by mKlus** [comment]
@wenbozhu could you please give me access to https://github.com/grpc/grpc-web/ as well?

**Comment by floridoo** [comment]
@wenbozhu I am also interested in early access. Thank you!

**Comment by hamiltop** [comment]
Would like to take a look and get early access.

**Comment by wenbozhu** [comment]
DONE.

@zinuga 

@ddevienne 
@kenkeiter 
@pkieltyka 

Some background about this whole white-listing thing: 

grpc-web was originally developed for internal use (excluding nginx), and we are still going through the internal process before we can make the repo open. 

nginx doesn't support http/2 upstream (expected soon), which means any existing nginx users will not be able to use nginx as a reserve-proxy (for load balancing) at the same time.

Our current goal is to make the repo open in early Q2 (as alpha). At the same time, we will ship nginx with LB support.

Thanks for your interests. And your actual feedback is much appreciated too.

**Comment by NoahAndrews** [comment]
I'd appreciate early access.

**Comment by rjarmstrong** [comment]
Please could I get early access? Looking at building a chrome extension.

**Comment by vsimon** [comment]
I'm also interested in early access! Thanks!

**Comment by chiahan1123** [comment]
@wenbozhu Could I please get early access to grpc-web? Would like to experiment with it on my web application. Thanks!

**Comment by snorecone** [comment]
@wenbozhu I'm interested in testing out early access to grpc-web, could I please be added? Cheers.

**Comment by ben-whitney** [comment]
I'm interested in trying it out. Can I have pre public release access?

**Comment by johanbrandhorst** [comment]
@wenbozhu could I trouble you for early access to grpc-web as well? Thanks!

**Comment by biolee** [comment]
@wenbozhu early access to grpc-web request Thanks!

**Comment by arbarlow** [comment]
@wenbozhu would love access, helping me to support web for https://github.com/lileio

**Comment by kwc** [comment]
@wenbozhu access, please

**Comment by silas** [comment]
I'd appreciate early access.

**Comment by lutianming** [comment]
@wenbozhu early access please

**Comment by athyuttamre** [comment]
I'm teaching a class on distributed systems with gRPC. Could I get early access please?

**Comment by wenbozhu** [comment]
DONE.

**Comment by bram-ivs** [comment]
@wenbozhu is it possible to also have access to grpc-web. Thanks!

**Comment by sovchinn** [comment]
@wenbozhu could you please grant early access for me as well? Thank you!

**Comment by ekostem** [comment]
@wenbozhu interested in experimenting as well, thanks!

**Comment by xiaodongw** [comment]
@wenbozhu can i have the early access? thanks.

**Comment by StepFischer** [comment]
@wenbozhu Is it possible to get access to the grpc-web project?

**Comment by chansen-va** [comment]
@wenbozhu Would appreciate read access to the grpc-web project, thanks!

**Comment by hanechris** [comment]
@wenbozhu I would like to test grpc-web if possible, thanks in advance!

**Comment by subfuzion** [comment]
@wenbozhu Our team (including @freignat91 @ neha-viswanathan) would definitely appreciate access. We are preparing for DockerCon and have a CLI for our gRPC-based backend, but would love to be able to demo a simple web UI as well. And thanks for connecting on LI!

**Comment by karolszmaj** [comment]
@wenbozhu Would appreciate access to the grpc-web early access, thanks!

**Comment by wenbozhu** [comment]
DONE

**Comment by larryzhouatt** [comment]
@wenbozhu Would appreciate access to the grpc-web early access, thanks!

**Comment by mickeyreiss** [comment]
@wenbozhu I'm interested as well!

**Comment by Randigtschackspel** [comment]
@wenbozhu I'd love to take a look at the grpc-web as well!

**Comment by SamiHiltunen** [comment]
@wenbozhu could you give me access to grpc-web as well? Thanks in advance!

**Comment by ieugen** [comment]
Could you please give me access to the repo? I wish to test it for possible production usage in the future.

**Comment by fnbushes** [comment]
@wenbozhu very interested in having a look as well, could you give access?

**Comment by prince-mishra** [comment]
Evaluating grpc for possible prod usage; web component is a must; Can I get early access please?

**Comment by rrichardson** [comment]
Thanks for working on this! It's clear that it is in high demand.  Can I get access to the repo as well? Thanks.

**Comment by pankaj9um** [comment]
@wenbozhu can you please grant me read access as well?

**Comment by nornagon** [comment]
@wenbozhu I'd like to evaluate grpc-web also please!

**Comment by vyshane** [comment]
@wenbozhu can I please have access to the repo?

**Comment by trevex** [comment]
@wenbozhu  I would also like to get access to evaluate grpc-web for future use. In case the open alpha is close by feel free to ignore this request.

**Comment by SonarBeserk** [comment]
@wenbozhu I wish to take a look at grpc-web as well.

**Comment by bitjson** [comment]
@wenbozhu could you add me as well?

**Comment by ggprod** [comment]
@wenbozhu hi, I'd also like to try out grpc-web, could I get access also?  Thanks!

**Comment by Chumper** [comment]
@wenbozhu I would also like to have access if possible. Thank you!

**Comment by puffmaster** [comment]
may I have access as well?

**Comment by sdotz** [comment]
Add me to the access list please!

**Comment by snrostov** [comment]
@wenbozhu could you please give me access as well. Thanks!

**Comment by suitupalex** [comment]
@wenbozhu may I have access as well please?

**Comment by breznik** [comment]
@wenbozhu Looking for early access as well. Please and thank you~!

**Comment by inetfuture** [comment]
@wenbozhu I would like to try it too, please.

**Comment by xieepp** [comment]
@wenbozhu I am very interested, may I  please?

**Comment by paulreimer** [comment]
I would like to take a look at this as well, if you could please add me to the early access list.

Also a question:
IIUC, browser support for native gRPC is continuing still, although not generally available (e.g. https://bugs.chromium.org/p/chromium/issues/detail?id=422958) and native gRPC should be preferred over the grpc-web protocol in any browsers where it has landed?

So gRPC-Web is intended as a polyfill until gRPC is more fully supported? Is the design that gRPC-Web will use the regular gRPC protocol, if the browser supports it? Or would that be up to us -- to just use the generated Node bindings directly -- if gRPC support is detected?

**Comment by 54nch4** [comment]
@wenbozhu Could you give me access? please

**Comment by birju-s** [comment]
@wenbozhu   Have to present a talk on gRPC --> Kindly allow me to get early access to grpc-web.

**Comment by SonarBeserk** [comment]
@birju-s It is very alpha, not sure you would really be wise to go spreading it when the implementation isn't finished.

**Comment by birju-s** [comment]
Sure, I will mention that people on the web working on it. Will not announce about our initiative. Does it makes sense ?  @SonarBeserk

**Comment by wenbozhu** [comment]
DONE

===

@paulreimer 

The eventual goal is to support the native grpc protocol, and make the deployment of grpc-web effortless. The latter includes adding production-level grpc/grpc-web support to popular proxies, frameworks. 

What we don't plan to do is to add web-only features to the native grpc protocol, to keep the latter stable/efficient.  Example features include: compression, cors, xss/xsrf, custom message formats, keep-alive etc.

**Comment by jupp0r** [comment]
@wenbozhu - can you add me to the early access list for grpc-web, please? Thanks!

**Comment by paulreimer** [comment]
@wenbozhu that's great. Now that I've seen the example nginx proxy implementation adding the `grpc_pass` directive, I can see what you mean about adding support to existing proxies (e.g. this is an example of what an eventual upstream nginx module _could_ look like).

The samples worked great by the way, although I haven't looked much at perf/latency yet.

**Comment by paulreimer** [comment]
Another question about the scope of gRPC-Web spec; I see there is an alternate implementation that mentions gRPC-Web here:
https://github.com/improbable-eng/grpc-web

However at a glance, the JS API is not the same (and seems to be different from the Node gRPC JS API as well...), but I assume that it implements the same transport/framing protocol described here. Does this spec (gRPC-Web) aim to provide a standard JS API (e.g. the official gRPC Node API), or is your implementation just a reference design for what a JS API might look like (and the @improbable-eng repo is another possible -- but different -- implementation of the spec).

**Comment by lobiCode** [comment]
@wenbozhu Add me too, please

**Comment by tamird** [comment]
@wenbozhu @zinuga we'd like early access to grpc-web. We may experiment with it in CockroachDB.

**Comment by nartamonov** [comment]
@wenbozhu could you give me early access to grpc-web? Thanks! )

**Comment by grapemix** [comment]
@wenbozhu We would like to have early access to grpc-web too if seats are available. We already have a few ionic2 clients. Thanks.

**Comment by Brahmasmi** [comment]
@wenbozhu Could you please grant access to the grpc-web repository? Thanks.

**Comment by cartazio** [comment]
i'd like to request read access too please :)

**Comment by nexeck** [comment]
I would like to request access, too :)

**Comment by miseri** [comment]
@wenbozhu Please grant me access too. Thanks!

**Comment by odino** [comment]
@wenbozhu any update on the progress of grpc-web?

Cheers!

**Comment by viggyfresh** [comment]
Access for me too please!

**Comment by Kylex87** [comment]
@wenbozhu I would like to request access as well. Thanks!

**Comment by fhofmann** [comment]
please add me as well, thanks!

**Comment by wenbozhu** [comment]
DONE

===


@paulreimer 

We are committed to supporting the gRPC Node API, including a future TS version. The client library is already used in various products, so we will keep the API as stable as possible, while any future changes will be made backward compatible.

The protocol spec is subject to change, inc. new features. We are actively working on a new version of Nginx gateway with Nginx providing the HTTP/2 support (to backend servers).  In addition to the Nginx gateway, we expect to see other gateways or frameworks to support the grpc-web protocol, e.g. Envoy, go-middleware from Improbable, with full or partial features expected for a gRPC client library.

**Comment by floriangoeldi** [comment]
Hi, please add me as well. Thanks in advance!

**Comment by drewbard** [comment]
@wenbozhu can you add me as well? Thanks in advance!

**Comment by myfancypants** [comment]
@wenbozhu I would love a chance to access the repo. Many thanks in advance!

**Comment by Boscop** [comment]
@wenbozhu Please add me too, thanks!

**Comment by selckin** [comment]
I would like to request access as well. Thanks!

**Comment by llchan** [comment]
@wenbozhu please add me too, thanks!

**Comment by hhclam** [comment]
Hi! I would like to request access to this as well. I'm trying to promote the use of gRPC in my company and one blocker is use of gRPC in the browser.

**Comment by csboling** [comment]
@wenbozhu I'm also interested in access. Thanks!

**Comment by MichaelTang1980** [comment]
@wenbozhu I'm also interested in access. Thanks!

**Comment by alexdcox** [comment]
@wenbozhu May I have access too please? Thanks in advance.

**Comment by marcuswestin** [comment]
@wenbozhu Me too :)

**Comment by wstrange** [comment]
Me too ...

(It might just be easier to make the repo read only to the world?)

**Comment by pieterlouw** [comment]
@wenbozhu 
Can I have access please?

**Comment by david7482** [comment]
@wenbozhu
I am also have interested to the exciting grpc-web. Thanks!!!

**Comment by alemoreau** [comment]
@wenbozhu May I have access too ? Thanks

**Comment by patrickwmcgee** [comment]
@wenbozhu May I have access as well?

**Comment by HTChang** [comment]
@wenbozhu I am also interested in grpc-web
May I have access too ? 
Thanks

**Comment by JamesGX** [comment]
@wenbozhu  I would like to have access as well. May I have it? Thanks.

**Comment by FrankChung** [comment]
I would like to request access as well. Thanks you!

**Comment by jellyhorn** [comment]
@wenbozhu May I have access as well? It is good to have grpc in my web application. Thanks!

**Comment by chiaen** [comment]
@zinuga @wenbozhu I really like to try the grpc-web. Please grant me the access. Thank you!

**Comment by xaviiic** [comment]
please add me as well, thanks.

**Comment by sheeley** [comment]
could i have access as well? thanks!

**Comment by groldan** [comment]
I'm working on the networking aspects of geogig (a git like DVCS for geospatial data) and would love access to the early spec/impl. Can I be whitelisted please?

**Comment by jgrossophoff** [comment]
Would love to have access as well =)

**Comment by HTChang** [comment]
didn't get the permission for grpc-web after two days, @zinuga @wenbozhu can you help me?

**Comment by wenbozhu** [comment]
DONE. 

Thanks for all the interests.  Due to changes in the proxy layer, we may have to delay the public alpha to June or later. 

Please let us know your feedback on github.com/grpc/grpc-web.

**Comment by ildar-samit** [comment]
@wenbozhu - I'd love access as well please. Thanks!

**Comment by borisceranic** [comment]
@wenbozhu - Any chance for me to get early access, too? Thanks!

**Comment by marcoflowers** [comment]
Same as above.

**Comment by CMon** [comment]
@wenbozhu - I would like to have access to it as well. Thanks in advance.

**Comment by magiccrafter** [comment]
@wenbozhu would love to get early access to grpc-web. thanks!

**Comment by cemizm** [comment]
Would love to have access as well 👍

**Comment by dsudia** [comment]
I would love to have access. Put in on the google form on the grpc site but was just told about this. thanks!

**Comment by aragozin** [comment]
@wenbozhu - I would like to have access to it as well. Thanks in advance.

**Comment by msalman86** [comment]
@wenbozhu, could I please get access as well? thank you

**Comment by franzwilhelm** [comment]
@wenbozhu Could you please add me as well. Thanks!

**Comment by paul-barton** [comment]
@wenbozhu please add me as well. Side question regarding a separate repo that references the spec [improbable-eng/grpc-web](https://github.com/improbable-eng/grpc-web), is this the initial ts implementation of the spec or something that project owner decided to implement on their own?

**Comment by kschima** [comment]
@wenbozhu  could I please get access as well? thank you

**Comment by ysaakpr** [comment]
could I get read permission to grpc-web?

**Comment by Lonelind** [comment]
That's exactly what I was looking for! Could I please get access? Thank you!

**Comment by demongaorui** [comment]
@wenbozhu

I would like to have access to it as well. Thanks in advance.

**Comment by avidmouse** [comment]
@wenbozhu
can you add me to pls:)

**Comment by schmich** [comment]
@wenbozhu grpc-web access would be appreciated, thanks!

**Comment by pjschlic** [comment]
@wenbozhu  Can I get early access to the github.com/grpc/grpc-web repo also?

**Comment by Grey2k** [comment]
@wenbozhu Can I get early access to the github.com/grpc/grpc-web repo also?

**Comment by ksaldana1** [comment]
@wenbozhu Would love access to the grpc-web repo. Thanks!

**Comment by wenbozhu** [comment]
@paul-barton  It's a separate implementation, but we have worked together to ensure spec compliance. The TS version of the grpc/grpc-web client is not yet open-sourced.

**Comment by chenzhekl** [comment]
@wenbozhu Would be appreciated to have read access to the grpc-web repo. Thanks!

**Comment by althk** [comment]
@wenbozhu can you please grant me read access to the grpc-web repo? Thanks!

**Comment by cpx86** [comment]
@wenbozhu Can I also have read access?

**Comment by Meronat** [comment]
I would appreciate access, thanks!

**Comment by danielgeri** [comment]
@wenbozhu could i have access to grpc-web as well? :)

**Comment by fixed-point** [comment]
@wenbozhu Could I please get read access to the grpc-web repo? Thanks! :)

**Comment by Fohlen** [comment]
can we at @inexorgame test this as well? :+1:

**Comment by geoffmiller** [comment]
@wenbozhu me too please and thank you.

**Comment by ehaydenr** [comment]
@wenbozhu, could I please get access to the grpc-web repo as well? Thanks.

**Comment by jeongoh** [comment]
@wenbozhu, please let me have the access to the repo too. Thanks in advance!

**Comment by joshua** [comment]
@wenbozhu can i get access as well. thanks!

**Comment by varshanbp** [comment]
@zinuga @wenbozhu Neeed Early access to GPRC Web. Please whitelist my ID too.

**Comment by justinsb** [comment]
@zinuga @wenbozhu can I please get access also

**Comment by jaytaylor** [comment]
@zinuga @wenbozhu I'm also keen on getting access :)

**Comment by adieu** [comment]
@zinuga @wenbozhu can I get early access as well. thanks a lot.

**Comment by wiktortomczak** [comment]
@wenbozhu Could you please give me access? We have developed a simplistic browser JS gRPC client while waiting for the official implementation and we would like to make sure it doesn't diverge too much, to make future porting easy.

**Comment by jaytaylor** [comment]
@wiktortomczak Have you seen https://github.com/improbable-eng/grpc-web/ ? It seems like a pretty good starting place.

**Comment by wiktortomczak** [comment]
@jaytaylor Thanks for the pointer. How is https://github.com/improbable-eng/grpc-web/ related to the official implementation?

**Comment by jprokop-va** [comment]
@wenbozhu I'm also interested in access to grpc-web, please.

**Comment by jaytaylor** [comment]
@wiktortomczak Not sure about "official", but unofficially I can say that the grpc-web works great for sharing protobufs and RPCs between golang gRPC server <=> client browser javascript.

**Comment by jaytaylor** [comment]
@wiktortomczak Not sure about "official", but unofficially I can say that grpc-web has worked great for sharing protobufs and RPCs between golang gRPC server <=> client browser javascript.

**Comment by nizsheanez** [comment]
@wenbozhu please grand for me Early access to GPRC Web

**Comment by slamdev** [comment]
@wenbozhu I'm also interested in access to **grpc-web**

**Comment by ivantopo** [comment]
hey @wenbozhu, I also would like early access to grpc-web, thanks in advance!

**Comment by YPhyu** [comment]
@wenbozhu Could I also get an access to grpc-web. Thank you.

**Comment by teddy-dubal** [comment]
@zinuga @wenbozhu can I get early access as well. thanks a lot.

**Comment by duncup** [comment]
@wenbozhu @zinuga   Could I also get an access to grpc-web. Thank you.

**Comment by Jordan-Hall** [comment]
@wenbozhu @zinuga Could I get early access to grpc-web. Thank you in advance

**Comment by agsimeonov** [comment]
@wenbozhu @zinuga I would love early access to grpc-web as well. Thank you so much for the hard work.

**Comment by Disturbing** [comment]
@wenbozhu @zinuga  please provide early access, I'd love to contribute! Thank you so much for making grpc-web!

**Comment by wenbozhu** [comment]
@wiktortomczak   as long as your implementation follows the Node API, future migration should be easy

**Comment by mmmveggies** [comment]
@wenbozhu @zinuga Humbly requesting early access to grpc-web. Thanks!

**Comment by bgphipps** [comment]
would love to have access to grpc-web as well, thanks!

**Comment by pedrocelso** [comment]
May I have early access as well?

**Comment by rsteinmetz** [comment]
I would love early access too please! Thank you!

**Comment by AJCStriker** [comment]
I'd love to get early access on this if it's still available!

**Comment by LinusDev-Niall** [comment]
@wenbozhu If you are still doing early-access I would be very interested in getting involved as at work we are looking at ways to port our existing products to a golang codbase. The ability to access grpc from the browser is something that has been getting in the way with migrating a few key products.

**Comment by aymericbeaumet** [comment]
I would also be interested in early access!

**Comment by josham** [comment]
@wenbozhu @zinuga I'm interested in early access to grpc-web, thanks!

**Comment by hitsubunnu** [comment]
@wenbozhu

I would like to have access to it as well. Thanks in advance.

**Comment by chengkunzzz** [comment]
@wenbozhu 
I’m so interested in the early access, thanks very much!

**Comment by serche** [comment]
@wenbozhu
Please, I'd like to have early access to grpc-web. Thanks.

**Comment by Martaver** [comment]
@wenbozhu we're building a web facing commercial product that I believe would benefit from GRPC. I'd like to investigate. Can you provide me with access to grpc-web? Thanks!

**Comment by faceace** [comment]
@wenbozhu would you please add me to grpc-web too.

**Comment by JulCesWhat** [comment]
@wenbozhu 
I am really interested in the grpc-web. Could I get early access to it? Thank you!

**Comment by RenatoUtsch** [comment]
@wenbozhu 
Could I please have access to grpc-web? I'm interested in testing it and would like to have early access to it.

**Comment by dsojevic** [comment]
@wenbozhu I'm also interested in early access to grpc-web if you'd please add me too. Thanks!

**Comment by fabricedesre** [comment]
@wenbozhu
I’m also interested in getting early access, thank you very much!

**Comment by sarelp** [comment]
@wenbozhu @zinuga I'm interested in early access to grpc-web, thanks!

**Comment by asaarnak** [comment]
@wenbozhu @zinuga I'm also interested to use the early access to grpc-web, thanks!

**Comment by c100k** [comment]
@wenbozhu Hi ! I'm starting a new project so I'd really like to experiment it as well and give feedback. Thanks !

**Comment by dabfleming** [comment]
@wenbozhu Currently looking into options for this and would appreciate early access to this as well, thanks.

**Comment by k2wanko** [comment]
@wenbozhu Please add early access to grpc-web.
Thanks

**Comment by YamiOdymel** [comment]
After a year, and I still have no idea about the current status of the grpc-web. And yet, people are still commenting `Add me please`, really?

This is an issue, not a Google Form.

**Comment by telekid** [comment]
@wenbozhu I'm also interested in receiving access to grpc-web.

**Comment by sebglon** [comment]
@wenbozhu @zinuga Please add early access to grpc-web.
Thanks

**Comment by bonnyr** [comment]
@wenbozhu @zinuga Please add early access to grpc-web.
Thanks

**Comment by seanchann** [comment]
@wenbozhu @zinuga Please add early access to grpc-web.
Thanks

**Comment by r3h0** [comment]
@wenbozhu @zinuga could I please get early access to grpc-web? Thanks!

**Comment by tyler-sommer** [comment]
What exactly would be lost if the `grpc-web` repo was made public? Just disable issues except by maintainers or contributors if you're worried about noise. It doesn't make sense IMO to keep it pseudo-private except for anyone who asks for access-- might as well just let people see it.

**Comment by wenbozhu** [comment]
@tyler-sommer 

I guess we do owe an explanation on this. grpc-web was initially developed for google's own projects and as a result, the development work all happens internally (still today). We are yet to go through the process to open-source the code base (inc. tests, inter-op, docs), and we are actively working on finishing the process (asap).

The purpose of the whitelist (aka early-access) was to gather as much feedback as possible, such as APIs, the use of closure compiler. We will send out a user survey soon for this.

@hsaliak

**Comment by tyler-sommer** [comment]
@wenbozhu Good to know, thanks for the insight! I know I'm looking forward to getting my hands on grpc-web. Looks like many others are too, so I'm glad that it seems the ball is still rolling toward a public release.

**Comment by hsaliak** [comment]
@tyler-sommer it is - we want to get this out as soon as possible and do as much development as we can on this in the open. As @wenbozhu mentioned, we have some process to go through and at the same time, take the opportunity to collect some indepth feedback from early adopters to make the project better

**Comment by drausin** [comment]
@wenbozhu, thank for the update. I'd also like early access, if still possible. Thanks!

**Comment by erikperttu** [comment]
Was checking the /grpc/grpc-experiments/ thread but could not see any updates, I guess I was monitoring the wrong thread. Is it possible for me to get an invite as well @wenbozhu ? Thanks in advance!

**Comment by main** [comment]
@wenbozhu @zinuga Please give me early access to grpc-web.

**Comment by Carmageddon** [comment]
Please give me early access as well @zinuga @wenbozhu

**Comment by mahtuag** [comment]
@zinuga @wenbozhu could I also get an early access to grpc-web? I'd like to evaluate it over the weekend.

**Comment by grivy** [comment]
@zinuga @wenbozhu could I get early access to grpc-web as well? Thanks!

**Comment by ryanjaeb** [comment]
@wenbozhu @zinuga I would also appreciate early access.  Thank you!

**Comment by debudda** [comment]
@wenbozhu @zinuga Hi! I would like to receive an early access too. Thanks in advance!

**Comment by vahid-sohrabloo** [comment]
Please give me early access as well @zinuga @wenbozhu

**Comment by tok-kkk** [comment]
Could I have early access? @wenbozhu  @zinuga  Thanks in advance.

**Comment by eschgi** [comment]
@zinuga @wenbozhu Could I have also early access? Thanks in advance.

**Comment by jsmouret** [comment]
@zinuga @wenbozhu You must receive quite a lot of spam :) Could I get early access too? Thanks a lot

**Comment by stefando** [comment]
Please give me early access as well @zinuga @wenbozhu

**Comment by Sgiath** [comment]
@zinuga what is the status on grpc-web? I see it being referenced as EA a year ago and it is still not generally available.
Also can I get access too?

**Comment by wenbozhu** [comment]
DONE. 

@Sgiath  soon for the public version.  Speaking of the delay, we had been waiting for Nginx http/2 upstream support, which is still not there.  We plan to just go ahead with what we have now, and rely on grpc-lb + Nginx to work in future.

@stanley-cheung

**Comment by Skisas** [comment]
Could I also receive early access @zinuga @wenbozhu

**Comment by SergejIsbrecht** [comment]
Could I also receive early access? @zinuga @wenbozhu

**Comment by radu-matei** [comment]
Any news here?
Would really like to take a look at it, @zinuga, @wenbozhu 

Thanks,
Radu M

**Comment by michaeljohnbennett** [comment]
Yes, can you ping me once its available for testing please?

**Comment by vahdet** [comment]
Well, early access is also welcome here. Otherwise, I am gonna stick with REST when my heart beats here. And it would make me sick :/
So yes, early access for me too please.

**Comment by lifk** [comment]
Is the early access still ongoing? If so I would like an invite. I'm trying to hack a custom proxy for gRPC using js but I would prefer an official implementation

**Comment by IanDuncanson** [comment]
@wenbozhu @zinuga Could I receive early access as well please? Thanks in advance.

**Comment by IanDuncanson** [comment]
@wenbozhu @zinuga My access to grpc-web was removed because I didn't have 2FA enabled. I have enabled this now. Could you re-add me to the early access program? Thanks in advance.

**Comment by wenbozhu** [comment]
@IanDuncanson and everyone else

Due to some recent policy changes, you now need enable 2FA (Two-factor Authentication) in order to retain the early-access permission to the grpc/grpc-web repo. 

You can check your 2FA status via  Github.com > Settings > Security

If you lose access once enforcement begins, you can have your access reinstated fully if you enable 2FA on your account within 3 months. After 3 months, your ACL entries will be purged.

If you need any assistance or have any issue, please email [me](mailto:web@google.com) directly.

Sorry for the inconvenience.

***

We expect to make the grpc-web repo public by next month, which requires going through Google's open-source process. Currently all the grpc-web development is being done internally.

**Comment by Luiz0tavio** [comment]
@wenbozhu @zinuga Could I receive early access as well please? 
Already enabled 2FA. Thanks in advance.

**Comment by topu** [comment]
@wenbozhu @zinuga  Could I also receive early access? Thanks in advance.

**Comment by masonforest** [comment]
Could I have early access as well?

I have 2FA enabled.

Thanks in advance! 😃

**Comment by nsriramulu** [comment]
@wenbozhu I would also like early access please :)

**Comment by DrQwertySilence** [comment]
I Can Has ~~Cheezburger~~ early access please?
Thanks in advance.

**Comment by adamyi** [comment]
Hi @wenbozhu 
Can I please have early access as well? Thanks a lot!

**Comment by vahid-sohrabloo** [comment]
@wenbozhu @zinuga Could I receive early access as well please?
Already enabled 2FA. Thanks in advance.  :smile:

**Comment by akbars38** [comment]
Could I have early access as well?

I have 2FA enabled.

Thanks in advance!

**Comment by hexfusion** [comment]
@wenbozhu I am interested in early access, please advise. 2FA is enabled, thank you.

**Comment by trivigy** [comment]
@wenbozhu @zinuga I would also appreciate early access. Thank you!

**Comment by biolee** [comment]
Could I have early access as well?

I have 2FA enabled.

Thanks in advance!

**Comment by shell909090** [comment]
@wenbozhu 
    May I have early access? I have 2FA enabled.
    Thx.

**Comment by benitogf** [comment]
@wenbozhu @zinuga can I have early access? 2FA is enabled, thank you!

**Comment by wenbozhu** [comment]
Thanks for all the interests. 

If you have tried grpc-web, please help fill this [short survey](https://docs.google.com/forms/d/15Yh6ZPT5djjC0GfEr3GMjw7A3C1lUsbdhH08oDVdw3U/edit). We are very close to making the repo public ... and finally close this sign-up thread ;)

**Comment by zhshr** [comment]
Can I have early access? I filled the form and have 2FA enabled weeks ago but no changes. THanks.

**Comment by arjunyel** [comment]
@wenbozhu I have 2fa enabled, I would love access so I can start a project with it ASAP :)

**Comment by cloudspinner** [comment]
@wenbozhu Could I have early access? Thanks in advance.

**Comment by drew-moore** [comment]
@wenbozhu @zinuga I too have 2fa enabled and would love to have early access! Our team is actively putting together POC's for bringing gRPC to the frontend and we're very curious what this will look like.

**Comment by tmthrgd** [comment]
@wenbozhu I hate to add even more noise to this issue, but I'd love early access if it's still open. (2fa is active).

**Comment by paulopereiraptc** [comment]
Hello,
could I also get an early access to grpc-web? 
Best regards

**Comment by haebou** [comment]
Hello all. 
If grpc-web is still being vetted before going public, and you're still whitelisting accounts for early access, could I please get access as well? 2FA enabled.

**Comment by zxtwonder** [comment]
Hi, I am directed here from today's community meeting. Can I get whitelisted for grpc-web?

We would like to evaluate feasibility of using gRPC to replace RESTful APIs from web apps. Thank you!

**Comment by dpadliya** [comment]
Hi - @wenbozhu @zinuga Can I please get whitelisted?  I'd like to test out a grpc client.

**Comment by miensol** [comment]
Hi @wenbozhu  @zinuga. Could you please whitelist me too? We're doing lots of small projects and I'm eager to replace some of our RPC over REST APIs with grpc.

**Comment by natronq** [comment]
@wenbozhu @zinuga 
Could you please enable me for early access? Thank you in advance.

**Comment by sharkman** [comment]
@wenbozhu @zinuga please whitelist me as well if it is possible

**Comment by MHDante** [comment]
@wenbozhu @zinuga I would love to have access as well.

**Comment by dbarzdys** [comment]
Hi @wenbozhu Could I also get an early access?
I have 2FA enabled.

Thanks  :smile:

**Comment by fizz** [comment]
I've enabled 2fa and I've filled out the form; could I please have access? Thank you!

**Comment by scottwio** [comment]
Could I have early access as well?

I have 2FA enabled.

**Comment by xsorifc28** [comment]
Could I get early access please?
I have enabled 2FA on my github account.

**Comment by L-Lavigne** [comment]
Hi @wenbozhu and @zinuga, I'd love to get early access as well. Got 2FA. Thanks in advance!

**Comment by jrbury** [comment]
Hi @wenbozhu and @zinuga I have filled out the form, have 2FA and am very interested in moving to gRPC from my current websocket setup if you're still accepting early access users!

Thanks!

**Comment by terborg** [comment]
@wenbozhu @zinuga Could I please have early access? I have 2FA enabled. Thank you!

**Comment by wenbozhu** [comment]
Making a short announcement here: 

[grpc/grpc-web](https://github.com/grpc/grpc-web) is now a public repo!

... which I hope officially concludes this 400+ issue thread ;)

If you are interested in road-map discussion or would like to contribute to this project, please check the [road-map doc](https://github.com/grpc/grpc-web/blob/master/ROADMAP.md).

Thanks!
